Added Support for creating a Zabbix Inventory. It is possible to tag the

argo1984 opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Added Support for creating a Zabbix Inventory. It is possible to tag the hosts and add Locations etc.
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
Module: zabbix_host

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
> ansible --version
ansible 2.3.0.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Mar 24 2017, 09:43:28) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
Usage Example in task:
<!--- Paste verbatim command output below, e.g. before and after your change -->
```
- name: "Create a new host or update an existing host's info"
  local_action:
    module: zabbix_host
    server_url: "{{ zabbix_url }}"
    login_user: "{{ zabbix_api_user }}"
    login_password: "{{ zabbix_api_pass }}"
    host_name: "{{ zabbix_hostname }}"
    host_groups: "{{ zabbbix_hostgroup }}"
    link_templates: "{{ zabbix_link_templates }}"
    inventory_mode: automatic
    inventory_zabbix:
      tag: "{{ your_tag }}"
      alias: "{{ your_alias }}"
      notes: "Special Informations: {{ your_informations | default('None') }}"
      location: "{{ your_location }}"
      site_rack: "{{ your_site_rack }}"
      os: "{{ your_os }}"
      hardware: "{{ your_hardware }}"
      vendor: "{{ your_vendor }}"
    status: "{{ zabbix_host_status }}"
    state: "{{ zabbix_create_host }}"
    proxy: "{{ zabbix_proxy }}"
    interfaces: "{{ zabbix_interfaces }}"
  tags:
    - api
```

gundalow commented on lib/ansible/modules/monitoring/zabbix_host.py:
```diff
@@ -90,6 +90,12 @@
+            - Add Facts for a zabbix inventory (e.g.: Tag)
+        required: false
+        default: None
+        version_added: '2.3'
```
`'2.4'`

argo1984 replied:
Hi,

i added an example. Sorry i didn't see your message before.

gundalow commented on lib/ansible/modules/monitoring/zabbix_host.py:
```diff
@@ -90,6 +90,12 @@
         required: false
         default: None
         version_added: '2.1'
+    inventory_zabbix:
```
Could you please add an example for this new option. Thanks

Commit:
Add Example

Commit:
Reference to example

Commit:
Fix error with collon (ansibot saw a yaml not a string)

gundalow commented:
@eikef Could you please cast your eye over this PR, thanks

D3DeFi commented:
Description of this option can use a link to [host inventory arguments](https://www.zabbix.com/documentation/3.2/manual/api/reference/host/object#host_inventory) as well as note that this needs to be paired with `inventory_mode: manual`. Same goes for example where it is used with `inventory_mode: automatic`.

Other than that it just needs [rebase](http://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html) due to merge conflicts and will be good to ship.

eikef commented:
Of the same opinion as @D3DeFi 

After a rebase it looks good to ship.

Commit:
Rebase with update of remote repository

Commit:
Change inventory mode to manual
add link to inventory documentation of zabbix

Commit:
Fix:
The test ansible-test sanity --test pep8 [?] failed with the following error:

lib/ansible/modules/monitoring/zabbix_host.py:532:1: E302 expected 2 blank lines, found 1
The test ansible-test sanity --test validate-modules [?] failed with the following error:

lib/ansible/modules/monitoring/zabbix_host.py:0:0: E309 version_added for new option (inventory_zabbix) should be 2.5. Currently 2.4

eikef commented:
Thank you :-)

eikef commented:
shipit

D3DeFi commented:
shipit
